x-pack/plugins/maps/common/content_management/v1/types.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  GetIn,
+  GetResult,
+  CreateIn,
+  CreateResult,
+  SearchIn,
+  SearchResult,
+  UpdateIn,
+  UpdateResult,
+  DeleteIn,
+  DeleteResult,
+} from '@kbn/content-management-plugin/common';
+import { MapContentType } from '../types';
+
+interface Reference {
+  type: string;
+  id: string;
+  name: string;
+}
+
+/* eslint-disable-next-line @typescript-eslint/consistent-type-definitions */
+export type MapAttributes = {
+  title: string;
+  description?: string;
+  mapStateJSON?: string;
+  layerListJSON?: string;
+  uiStateJSON?: string;
+};
+
+export interface MapItem {
```

Comment:
this seems like saved object type that could be abstracted as it will be the same for all our saved objects ? SavedObject<MapAttributes> ?